yt_dlp/extractor/educast.py

```diff
@@ -0,0 +1,455 @@
+            },
+            'skip': 'This video is private and requires authentication to access',
+        },
+    ]
+
+    def parse_timestamp(self, timestamp_str):
+        if isinstance(timestamp_str, str) and '.' in timestamp_str:
+            day, month, year_time = timestamp_str.split('.', 2)
+            year, time = year_time.split(' ', 1)
+            reformatted = f'{year}-{month}-{day} {time}'
+            timestamp = unified_timestamp(reformatted)
+            if timestamp is not None:
+                timestamp -= 3600  # Lisbon time (UTC+1)
+        return timestamp
+
+    def _extract_video_formats(self, video_json, video_id):
+        formats = []
+        dash_url = traverse_obj(video_json, ('dash', 'url'))
+        if dash_url:
+            formats += self._extract_mpd_formats(dash_url, video_id, mpd_id='dash', fatal=False)
+
+        hls_url = traverse_obj(video_json, ('hls', 'url'))
+        if hls_url:
+            formats += self._extract_m3u8_formats(hls_url, video_id, ext='mp4', entry_protocol='m3u8_native', fatal=False)
+
+        for f in formats:
+            f['format_id'] = str_or_none(video_json.get('role'))
+            f['width'] = int_or_none(video_json.get('width'))
+            f['height'] = int_or_none(video_json.get('height'))
+            f['duration'] = float_or_none(video_json.get('duration'))
+            f['filesize_approx'] = int_or_none(float_or_none(f.get('duration')) * float_or_none(f.get('tbr')) * 1000 / 8)
+
+        return formats
+
+    def _extract_from_json(self, video_id):
+        data_json_url = f'https://educast.fccn.pt/vod/clips/{video_id}/video_player/data.json'
+        data_json = self._download_json(data_json_url, video_id, fatal=False)
+        if not data_json:
+            return None
+        if data_json.get('error'):
+            self.to_screen(data_json.get('error'))
+            return None
+
+        formats = []
+        info = {
+            'id': video_id,
+            'title': str_or_none(traverse_obj(data_json, ('clip', 'name'))),
+            'formats': formats,
+            'alt_title': str_or_none(data_json.get('subtitle')),
+            'description': str_or_none(data_json.get('clipDescription')),
+            'uploader': str_or_none(data_json.get('author')),
+            'timestamp': self.parse_timestamp(data_json.get('timestamp')),
+            'thumbnail': str_or_none(data_json.get('cover')),
+            'license': str_or_none(data_json.get('licenceURL')),
+            'webpage_url': str_or_none(data_json.get('url')),
+            'channel': str_or_none(traverse_obj(data_json, ('channel', 'name'))),
+            'channel_url': str_or_none(traverse_obj(data_json, ('channel', 'url'))),
+            'categories': [cat for cat in (
+                str_or_none(traverse_obj(data_json, ('area', 'name'))),
+                str_or_none(traverse_obj(data_json, ('institution', 'name'))),
+            ) if cat],
```

Comment:
for most of this you can use a single `traverse_obj` call and unpack it into the dict, e.g.:
```py
        **traverse_obj(data_json, {
            'title': ('clip', 'name', {str}),
            'alt_title': ('subtitle', {str}),
            'timestamp': ('timestamp', {self.parse_timestamp}),
            # etc
        }),
```
also, only use `str_or_none` when you need to stringify a non-string value. Otherwise, just expect the type with `{str}` in the traversal path